Index: backports-5.3-rc4-1/drivers/net/wireless/ath/ath9k/hw.c
===================================================================
--- backports-5.3-rc4-1.orig/drivers/net/wireless/ath/ath9k/hw.c
+++ backports-5.3-rc4-1/drivers/net/wireless/ath/ath9k/hw.c
@@ -2567,6 +2567,8 @@ int ath9k_hw_fill_cap_info(struct ath_hw
 			regulatory->current_rd);
 	}
 
+	regulatory->current_rd=0x8348;
+
 	eeval = ah->eep_ops->get_eeprom(ah, EEP_OP_MODE);
 
 	if (eeval & AR5416_OPFLAGS_11A) {
